### PR TITLE
Warnings signed unsigned

### PR DESCRIPTION
**Pull request by IngCr3at1on**
This is part of PR #229

### CODE REVIEW

**Comment by IngCr3at1on** [comment]
This has been rebased over master and updated to accommodate #407

File: src/addrman.cpp [hunk 1]
```diff
@@ -103,12 +103,11 @@ CAddrInfo* CAddrMan::Create(const CAddress &addr, const CNetAddr &addrSource, in
     return &mapInfo[nId];
 }
 
-void CAddrMan::SwapRandom(int nRndPos1, int nRndPos2)
+void CAddrMan::SwapRandom(unsigned int nRndPos1, unsigned int nRndPos2)
 {
     if (nRndPos1 == nRndPos2)
         return;
 
-    assert(nRndPos1 >= 0 && nRndPos2 >= 0);
```
Comment (mitchellcash) [inline src/addrman.cpp]: Is this meant to be removed?

Comment (IngCr3at1on) [inline src/addrman.cpp]: Yes, no, either really... They removed it from bitcoin but to be honest I don't see a problem with simply casting 0 to a unsigned integer and keeping it.

Comment (mitchellcash) [inline src/addrman.cpp]: Ahh right. Happy to leave it removed. Whichever you want :)

Comment (IngCr3at1on) [inline src/addrman.cpp]: Honestly I'm not sure why they removed it, it may be safer to retain it and cast the 0 as I said. I'm really not sure.

**Comment by IngCr3at1on** [comment]
This has been rebased over master do to merge conflicts.

**Pull request closed without merging**